system/ping6/ping6.c

```diff
@@ -92,8 +101,11 @@ static void show_usage(FAR const char *progname, int exitcode)
 
 static void ping6_result(FAR const struct ping6_result_s *result)
 {
+  FAR struct ping6_priv_s *priv = result->info->priv;
   char strbuffer[INET6_ADDRSTRLEN];
 
+  priv->code = result->code;
```

Comment:
i guess this way many of errors will be overwritten by ICMP_I_FINISH.
is it intended?

Reply:
Yes, you are right. Please review the updated version again.